spark-operator/src/main/java/org/apache/spark/k8s/operator/utils/PodPhase.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.utils;
+
+import io.fabric8.kubernetes.api.model.Pod;
+
+public enum PodPhase {
+  PENDING("pending"),
+  RUNNING("running"),
+  FAILED("failed"),
+  SUCCEEDED("succeeded"),
+  TERMINATING("terminating"),
```

Comment:
Although I understand why we have this here, It seems that we need an extra description because this is not one of the Pod phases according to K8s official definition.

- https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase
> Note: When a Pod is being deleted, it is shown as Terminating by some kubectl commands. This Terminating status is not one of the Pod phases.

Reply:
Please add more comments for `public enum PodPhase` and each values.